apps/src/code-studio/assets/SpriteLabAnimationUpload.jsx

```diff
@@ -0,0 +1,590 @@
+        availability: value || this.state.availability,
+        category: '',
+        metadata: '',
+        destination: '',
+        [ANIMATION_EXISTS_CHECK]: Status.WAITING
+      },
+      () => {
+        if (this.state.availability === AnimationLocations.level) {
+          this.determineIfAnimationAlreadyExists(
+            value,
+            filename.split('.')[0],
+            category,
+            fileData
+          );
+        }
+      }
+    );
+  };
+
+  handleCategoryChange = event => {
+    const value = event.target.value;
+    let {filename, availability, fileData} = this.state;
+    this.setState(
+      {
+        category: value,
+        metadata: '',
+        [ANIMATION_EXISTS_CHECK]: Status.WAITING
+      },
+      () => {
+        this.determineIfAnimationAlreadyExists(
+          availability,
+          filename.split('.')[0],
+          value,
+          fileData
+        );
+      }
+    );
+  };
+
+  handleAliasChange = event => {
+    const aliases = event.target.value?.split(',') || [];
+    let processedAliases = aliases.map(alias => alias.trim());
+    this.setState({aliases: processedAliases, metadata: ''});
+  };
+
+  // Set the upload status to indicate whether this file would override another
+  determineIfAnimationAlreadyExists = (
+    availability,
+    filename,
+    category,
+    fileData
+  ) => {
+    if (!fileData) {
+      return;
+    }
+    if (availability === AnimationLocations.library && category === '') {
+      this.setState({
+        [ANIMATION_EXISTS_CHECK]: Status.ALERT,
+        destination: 'No category selected'
+      });
+      return;
+    }
+    let {spriteLabLevelAnimations, spriteLabLibraryAnimations} = this.state;
+    let destination, animationExists;
+    switch (availability) {
+      case AnimationLocations.level:
+        destination = `level_animations/${filename}`;
+        animationExists = spriteLabLevelAnimations.includes(destination);
+        break;
+      case AnimationLocations.library:
+        destination = `category_${category}/${filename}`;
+        animationExists = spriteLabLibraryAnimations.includes(destination);
+        break;
+    }
+    this.setState({
+      [ANIMATION_EXISTS_CHECK]: animationExists
+        ? Status.ALERT
+        : Status.SUCCEEDED,
+      destination: destination
+    });
+  };
+
+  generateMetadata = () => {
+    const {filename, aliases, category} = this.state;
+    const metadata = {
+      name: filename.split('.')[0],
+      aliases: aliases,
+      frameCount: 1,
+      frameSize: this.state.frameSize,
+      looping: true,
+      frameDelay: 2,
+      categories: [category]
+    };
+    this.setState({metadata: JSON.stringify(metadata)});
+  };
+
+  animationExistsMessage() {
+    switch (this.state[ANIMATION_EXISTS_CHECK]) {
+      case Status.SUCCEEDED:
+        return 'This path and filename are available.';
+      case Status.ALERT:
+        return 'Filename already exists at this path. Please rename and try again.';
```

Comment:
Do we want to include 'rename and reupload'? I think that clarifies that the extra action here is to edit the file name locally and then re-start this upload process.

Reply:
Happy to add that. I kept it short to avoid needing to wrap to a second line in the UI, but I don't think there's a real reason we need to do that, especially at the cost of clarity.